### DISCUSSION TITLE
Seamless display patch

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1708] [Q&A]
Hello,

I am following [Kernel panic error on number of time reboot CPU:0, Error: cbb-noc@2300000, irq=15 L4T 35.4.1 - Jetson & Embedded Systems / Jetson Xavier NX - NVIDIA Developer Forums](https://forums.developer.nvidia.com/t/kernel-panic-error-on-number-of-time-reboot-cpu-0-error-cbb-noc-2300000-irq-15-l4t-35-4-1/287491)  to debug the exact error outlined in this post, with my NVIDIA NX GPU. I am off Kirkstone.
I believe I can add the "disable-seamless = <1>" where required in the dtb file, but regarding the second patch (Set the display power domains to always-on with the following patch), I am not sure how to add what is required as I don't have a/drivers/soc/tegra/powergate-bpmp.c .

Any general recommendation on how to map nvidia files with meta-tegra ?

Thank you.

**Comment by kekiefer** [comment]
The file [`drivers/soc/tegra/powergate-bpmp.c` exists in source tree for linux-tegra](https://github.com/OE4T/linux-tegra-5.10/blob/db857ecdbe10de229a3cf125831afd6226417eac/drivers/soc/tegra/powergate-bpmp.c), which is built by a recipe of the same name. You can clone the git repo and make a patch to bring into your append or use the `devtool` workflow in bitbake.

You may have to fix the suggested patch up a bit, the chip id TEGRA194 is defined with `#include <soc/tegra/fuse.h>` that I don't see in the patch.

Let us know if this fixes your problem. I ran into this issue as well, but I have a very specific setup on Xavier AGX with a custom board and displayport with a different configuration than the devkit, as well as a custom display that was very picky about the continuity of its signaling, so I chalked my experience and the additional changes I required up to personal problems.

**Reply by andregirard37** [reply]
Thanks for the help! 
Just to clarify, I need to clone the repo to make the patch only, correct ? I am a bit confused as to where can I find the .bb recipe for this. Is the name of the recipe linux-tegra-5.10 ? Is there a link to download the recipe ? For me to then add my patch in the .bbappend.

**Reply by kekiefer** [reply]
That's all correct. The recipe is in meta-tegra. You will add a bbappend in your layer that incorporates the patch you created.

You might find the devtool workflow easier. I prefer it, but some think it's more trouble than it's worth. https://wiki.yoctoproject.org/wiki/TipsAndTricks/Patching_the_source_for_a_recipe

Regarding testing the changes to kernel source, you may find you will need to build the `l4t-launcher-extlinux` recipe (assuming you're using extlinux, which will make this much easier) to get the combined kernel-initramfs image built. If you have package feeds set up, just update that package. Otherwise you should be able to build this recipe, and it will put the kernel-initramfs image in your deploy directory, which you can copy over by hand to the right place in /boot.
